include/revng/EarlyFunctionAnalysis/IRHelpers.h

```diff
@@ -0,0 +1,99 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "llvm/IR/Instructions.h"
+
+#include "revng/EarlyFunctionAnalysis/FunctionMetadata.h"
+#include "revng/Model/Binary.h"
+#include "revng/Model/IRHelpers.h"
+#include "revng/Support/Assert.h"
+#include "revng/Support/IRHelpers.h"
+#include "revng/Support/MetaAddress.h"
+#include "revng/TupleTree/TupleTree.h"
+
+namespace detail {
+
+inline TupleTree<efa::FunctionMetadata>
+extractFunctionMetadata(llvm::MDNode *MD) {
+  using namespace llvm;
+
+  efa::FunctionMetadata FM;
+  const MDOperand &Op = MD->getOperand(0);
+  revng_assert(MD != nullptr && isa<MDString>(Op));
+
+  StringRef YAMLString = cast<MDString>(Op)->getString();
+  auto MaybeParsed = TupleTree<efa::FunctionMetadata>::deserialize(YAMLString);
+  revng_assert(MaybeParsed);
+  MaybeParsed->verify();
+  return std::move(MaybeParsed.get());
+}
+
+} // namespace detail
+
+inline TupleTree<efa::FunctionMetadata>
+extractFunctionMetadata(const llvm::Function *F) {
+  auto *MDNode = F->getMetadata(FunctionMetadataMDName);
+  return detail::extractFunctionMetadata(MDNode);
+}
+
+inline TupleTree<efa::FunctionMetadata>
+extractFunctionMetadata(llvm::BasicBlock *BB) {
+  auto *MDNode = BB->getTerminator()->getMetadata(FunctionMetadataMDName);
+  return detail::extractFunctionMetadata(MDNode);
+}
+
+/// \brief Given a Call instruction and the model type of its parent function,
+///        return the edge on the model that represents that call, or nullptr if
+///        this doesn't exist.
+inline std::pair<efa::CallEdge *, MetaAddress>
```

Comment:
Maybe return a `const efa::CallEdge &`

Reply:
There's a case (line 58) in which I'm returning `return { nullptr, BlockAddress };`. I'm not sure, is there a way to return a CallEdge invalid if we want to return a reference?